### ISSUE TITLE
Missing scrobbles in Google Play Music

### ISSUE THREAD

**Issue by hornbuckle** [web-scrobbler/web-scrobbler#1883]
### Expected behavior

All songs that are recognized by Last.fm should scrobble when played.

### Current behavior

Similar to issue [#316](https://github.com/web-scrobbler/web-scrobbler/issues/316), this continues to be an issue in Chrome. Songs show as currently playing but then are frequently not saved to the play history when the song is completed. This has been an ongoing issue but does not seem to have any consistent behavior; a song that was previously skipped may scrobble the next time it is played. Songs that are clearly recognized and in the Last.fm database already just disappear from being scrobbled. I cannot remember the last time I was actually able to scrobble every song from an album as I listened from start to finish. At least one or two songs per album—sometimes more—always end up missing.

### Possible solution

I do not know exactly what the issue is, but it appears that tracks are just getting lost at random. I wonder if there are not enough attempts to actually send the track to Last.fm, and perhaps it only gets sent once, fails, and then does nothing until the next track starts playing. Tracks should be sent to Last.fm as many times as necessary until it is confirmed they have actually scrobbled.

### Steps to reproduce (for bugs)

Unfortunately, there are no exact steps to reproduce because the issue seems to happen at random. And whenever I actually have the DevTools up to debug the extension, it always magically scrobbles without issue.

### Steps for resolving the issue

- [ ] Reauthenticate on Last.fm

- [ ] Try to use unpacked extension from latest master

- [ ] Check other issues for duplicates

### Environment

* Version used: 2.13.0
* Browser name and version: Google Chrome 71.0.3578.98

**Comment by inverse** [comment]
Thanks for raising this. Have you observed this problem with any other sites or are you only using the extension with Google Play? 

Just to confirm you're seeing now playing notifications on last.fm successfully? but it's never saving upon completion? 

Would you be able to provide any logs the next time you see this? See: https://github.com/web-scrobbler/web-scrobbler/wiki/Debug-the-extension

**Comment by hornbuckle** [comment]
Hi, I really only use it for Google Play Music. And yes, I will see a song show up as "now playing" but then often (not "never") it does not save when going to the next song. Whenever I keep the debugger up, it always works! So unfortunately, I don't know how to provide more information, as it seems to be completely random.

For example, yesterday I was listening to an album from start to finish. Two of the songs out of 12 didn't save. When I went back and played them with the debugger open, then they saved!

**Comment by inverse** [comment]
Was this always an issue or a recent problem? It sounds frustrating either way. 

Lets see if anyone else can help with this issue or I can look when I have some free cycles.

**Comment by hornbuckle** [comment]
It has always been an issue. I only finally reported it when I realized someone else was having the same issue (although it seems to have resolved for them).

I am currently listening to an entire album and had a song get skipped. I had the debugger up, and here is what it showed:

```
Tab 13: New song detected: {
  "track": "the sweetest gift",
  "artist": "sade",
  "album": "lovers rock",
  "uniqueID": null,
  "duration": 301,
  "currentTime": 168,
  "isPlaying": true,
  "trackArt": "https://lh3.googleusercontent.com/CKytLFouLJB2eUfiiEE7MvQufnWQpya9S2Z8vrMrPtba9Hyyv4l-9CSyGFxx=s180-c-e100"
}
scrobble-service.js:90 Send "get info" request: 1
base-scrobbler.js:492 Last.fm: track.getinfo response:
<?xml version="1.0" encoding="UTF-8" ?>
<lfm status="ok"><track><name>The Sweetest Gift</name>
<mbid>f4ce3efa-a145-4247-b604-c4de90cc59da</mbid>
<url>https://www.last.fm/music/Sade/_/The+Sweetest+Gift</url>
<duration>139000</duration>
<streamable fulltrack="0">0</streamable>
<listeners>83865</listeners>
<playcount>307464</playcount>
<artist><name>Sade</name>
<mbid>67930b3e-e00b-469f-8c74-fd69f20522ec</mbid>
<url>https://www.last.fm/music/Sade</url>
</artist>
<album position="7"><artist>Sade</artist>
<title>Lovers Rock</title>
<mbid>5386e7b7-f4a1-3e5d-ad03-7608f3696bd9</mbid>
<url>https://www.last.fm/music/Sade/Lovers+Rock</url>
<image size="small">https://lastfm-img2.akamaized.net/i/u/34s/eced1e979c1ed3563db2568b9d1a13ab.png</image>
<image size="medium">https://lastfm-img2.akamaized.net/i/u/64s/eced1e979c1ed3563db2568b9d1a13ab.png</image>
<image size="large">https://lastfm-img2.akamaized.net/i/u/174s/eced1e979c1ed3563db2568b9d1a13ab.png</image>
<image size="extralarge">https://lastfm-img2.akamaized.net/i/u/300x300/eced1e979c1ed3563db2568b9d1a13ab.png</image>
</album>
<userloved>0</userloved>
<toptags><tag><name>female vocalists</name>
<url>https://www.last.fm/tag/female+vocalists</url>
</tag>
<tag><name>soul</name>
<url>https://www.last.fm/tag/soul</url>
</tag>
<tag><name>Sade</name>
<url>https://www.last.fm/tag/Sade</url>
</tag>
<tag><name>Smooth Jazz</name>
<url>https://www.last.fm/tag/Smooth+Jazz</url>
</tag>
<tag><name>jazz</name>
<url>https://www.last.fm/tag/jazz</url>
</tag>
</toptags>
</track>
</lfm>

coverartarchive.js:29 Using local/parsed artwork
controller.js:583 Tab 13: Song finished processing: {
  "parsed": {
    "artist": "sade",
    "track": "the sweetest gift",
    "album": "lovers rock",
    "uniqueID": null,
    "duration": 301,
    "currentTime": 168,
    "isPlaying": true,
    "trackArt": "https://lh3.googleusercontent.com/CKytLFouLJB2eUfiiEE7MvQufnWQpya9S2Z8vrMrPtba9Hyyv4l-9CSyGFxx=s180-c-e100"
  },
  "processed": {
    "artist": "Sade",
    "track": "The Sweetest Gift",
    "album": "lovers rock",
    "duration": 139
  },
  "internalId": "3acaced6af61e369620575504e374c50",
  "metadata": {
    "userloved": false,
    "startTimestamp": 1549323820,
    "connector": {
      "label": "Google Play Music",
      "matches": [
        "*://play.google.com/music/*"
      ],
      "js": [
        "connectors/googlemusic.js"
      ]
    },
    "artistThumbUrl": "https://lastfm-img2.akamaized.net/i/u/300x300/eced1e979c1ed3563db2568b9d1a13ab.png",
    "artistUrl": "https://www.last.fm/music/Sade",
    "trackUrl": "https://www.last.fm/music/Sade/_/The+Sweetest+Gift",
    "albumUrl": "https://www.last.fm/music/Sade/Lovers+Rock"
  },
  "userdata": {},
  "flags": {
    "isProcessed": true,
    "isScrobbled": false,
    "isCorrectedByUser": false,
    "isValid": true,
    "isMarkedAsPlaying": false,
    "isSkipped": false,
    "isReplaying": false
  }
}
controller.js:583 Tab 13: The song will be scrobbled in 150.5 seconds
scrobble-service.js:106 Send "now playing" request: 1
base-scrobbler.js:492 Last.fm: track.updatenowplaying response:
<?xml version="1.0" encoding="UTF-8"?>
<lfm status="ok">
  <nowplaying>
    <track corrected="0">The Sweetest Gift</track>
    <artist corrected="0">Sade</artist>
    <album corrected="0">lovers rock</album>
    <albumArtist corrected="0"></albumArtist>
    <ignoredMessage code="0"></ignoredMessage>
  </nowplaying>
</lfm>

controller.js:583 Tab 13: Song set as now playing
```
It says it would be "scrobbled in 150.5 seconds" but then goes back to "now playing" and never actually scrobbles? Is there anything useful in there? There was never a "Scrobbled successfully" message, but I don't know if that message always appears when a track saves.

The next line was just "Tab 13: New song detected: {" and then it started showing the next track, which did save.

Out the 11-track album, that is the only song that did not save.

**Comment by alexesprit** [comment]
@hornbuckle I use GPM too, and I noticed some time ago the website doesn't update track duration on track change if website is in an inactive tab, or if a browser is minimized. Seems like this happens in your case:
```
Tab 13: New song detected: {
  "track": "the sweetest gift",
  "artist": "sade",
  "album": "lovers rock",
  "uniqueID": null,
  "duration": 301, <----- parsed duration, i.e. taken from the website is 301 seconds.
  "currentTime": 168,
  "isPlaying": true,
  "trackArt": "https://lh3.googleusercontent.com/CKytLFouLJB2eUfiiEE7MvQufnWQpya9S2Z8vrMrPtba9Hyyv4l-9CSyGFxx=s180-c-e100"
}
```
```
controller.js:583 Tab 13: Song finished processing: {
  "parsed": {
    // same as `parsed` block listed above
  },
  "processed": {
    "artist": "Sade",
    "track": "The Sweetest Gift",
    "album": "lovers rock",
    "duration": 139 <----- duration received from external service (Last.fm database)
  },
```
The problem is the extension relies on parsed duration, but the extension gets it wrong sometimes due to GPM update limitations. So, in your case the correct duration could be 139 seconds, not 301 seconds.

Could you check and confirm if it happens when a browser is minimized, or a tab with GPM opened is inactive?

**Comment by hornbuckle** [comment]
@alexesprit Hi, yes, I can confirm that when tracks are missed, the tab using GPM is almost always inactive. Which is why when I'm actively checking for the issue and have the debugger open, the issue rarely (if ever) occurs.

**Comment by alexesprit** [comment]
@hornbuckle ok, thanks for the info.

**Comment by alexesprit** [comment]
Closed as a duplicate of #1588.
